bin/send-docker-hub-trigger.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# This script will trigger a multi-stage Docker build at the Docker hub repo referenced by $DOCKER_HUB_TRIGGER_URL.
+
+if [[ "$TRAVIS_PULL_REQUEST" == 'false' ]]; then
+  echo "Triggering Docker Hub build on branch @@BRANCH"
+  curl -H "Content-Type: application/json" --data '{"source_type": "Branch", "source_name": "@@BRANCH"}' -X POST "$DOCKER_HUB_TRIGGER_URL"
```

Comment:
Yep right here, instead of a `@@BRANCH` string that gets replaced, this should instead be either using `$TRAVIS_BRANCH` directly (and asserting it is present), or accepting a positional argument, e.g.:

``` bash
: "${TRAVIS_BRANCH:=${1}}"
[[ "${TRAVIS_BRANCH}" ]] || {
  echo "Missing TRAVIS_BRANCH" >&2
  exit 1
}

if [[ "${TRAVIS_PULL_REQUEST}" == 'false' ]]; then
# ...
```

Reply:
For some reason I wasn't able to get the Docker Hub trigger to work unless the `--data` was surrounded by single quotes, hence the torturous `sed` operation. I can take another look at this in a bit; maybe I'll have more luck with fresh eyes.